include/realtime_tools/realtime_box_best_effort.h

```diff
@@ -0,0 +1,220 @@
+   * @return false if the mutex could not be locked
+   * @note only safe way to access pointer type content (rw)
+  */
+  bool trySet(const std::function<void(T &)> & func)
+  {
+    std::unique_lock<std::mutex> guard(lock_, std::defer_lock);
+    if (!guard.try_lock()) {
+      return false;
+    }
+
+    func(value_);
+    return true;
+  }
+  /**
+     * @brief get the content with best effort
+     * @return std::nullopt if content could not be access, otherwise the content is returned
+    */
+  template <typename U = T>
+  [[nodiscard]] typename std::enable_if_t<!is_ptr_or_smart_ptr<U>, std::optional<U>> tryGet() const
+  {
+    std::unique_lock<std::mutex> guard(lock_, std::defer_lock);
+    if (!guard.try_lock()) {
+      return std::nullopt;
+    }
+    return value_;
+  }
+  /**
+   * @brief access the content (r) with best effort
+   * @return false if the mutex could not be locked
+   * @note only safe way to access pointer type content (r)
+  */
+  bool tryGet(const std::function<void(const T &)> & func)
+  {
+    std::unique_lock<std::mutex> guard(lock_, std::defer_lock);
+    if (!guard.try_lock()) {
+      return false;
+    }
+
+    func(value_);
+    return true;
+  }
+
+  /**
+     * @brief set the content and wait until the mutex could be locked (RealtimeBox behaviour)
+     * @return true
+    */
+  template <typename U = T>
+  typename std::enable_if_t<!is_ptr_or_smart_ptr<U>, void> set(const T & value)
+  {
+    std::lock_guard<std::mutex> guard(lock_);
+    // cppcheck-suppress missingReturn
+    value_ = value;
+  }
+  /**
+   * @brief access the content (rw) and wait until the mutex could locked
+  */
+  void set(const std::function<void(T &)> & func)
+  {
+    std::lock_guard<std::mutex> guard(lock_);
+    func(value_);
+  }
+
+  /**
+     * @brief get the content and wait until the mutex could be locked (RealtimeBox behaviour)
+     * @return copy of the value
+    */
+  template <typename U = T>
+  [[nodiscard]] typename std::enable_if_t<!is_ptr_or_smart_ptr<U>, U> get() const
+  {
+    std::lock_guard<std::mutex> guard(lock_);
+    return value_;
+  }
+  /**
+   * @brief get the content and wait until the mutex could be locked
+   * @note same signature as in the existing RealtimeBox<T>
+  */
+  template <typename U = T>
+  typename std::enable_if_t<!is_ptr_or_smart_ptr<U>, void> get(T & in) const
+  {
+    std::lock_guard<std::mutex> guard(lock_);
+    // cppcheck-suppress missingReturn
+    in = value_;
+  }
+  /**
+   * @brief access the content (r) and wait until the mutex could be locked
+   * @note only safe way to access pointer type content (r)
+   * @note same signature as in the existing RealtimeBox<T>
+  */
+  void get(const std::function<void(const T &)> & func)
+  {
+    std::lock_guard<std::mutex> guard(lock_);
+    func(value_);
+  }
+
+  // Make the usage easier by providing a custom assignment operator
+  // and a custom conversion operator
+  // Only to be used from non-RT!
```

Comment:
why not define this as docstring?

Reply:
Because this applies to all the following operators and not just the first one. I will separate it into a doc string per operator.